batch_tasks/resource/hook/batch_tasks.py

```diff
@@ -0,0 +1,385 @@
+                                'data': TASK_TYPE_ENUMERATOR_OPTIONS
+                            },
+                            {
+                                'label': 'Name',
+                                'type': 'text',
+                                'name': 'task_{0}_name'.format(index)
+                            }
+                        ]
+                    )
+
+                return {'success': True, 'message': '', 'items': items}
+            else:
+                # Create tasks on each entity
+                for entity_type, entity_id in entities:
+                    entity = session.get(entity_type, entity_id)
+                    for count in range(0, len(values.keys()) / 2):
+                        task_type = session.query(
+                            'Type where id is "{0}"'.format(
+                                values["task_{0}_typeid".format(count)]
+                            )
+                        ).one()
+
+                        # Get name, or assume task type in lower case as name.
+                        name = values["task_{0}_name".format(count)]
+                        if not name:
+                            name = task_type["name"].lower()
+
+                        # Query for existing task.
+                        query = (
+                            'Task where type.id is "{0}" and name is "{1}" '
+                            'and parent.id is "{2}"'
+                        )
+                        task = session.query(
+                            query.format(
+                                values["task_{0}_typeid".format(count)],
+                                name,
+                                entity["id"]
+                            )
+                        ).first()
+
+                        # Create task.
+                        if not task:
+                            task = session.create(
+                                "Task",
+                                {
+                                    "name": name,
+                                    "type": task_type,
+                                    "parent": entity
+                                }
+                            )
+
+                return {
+                    'success': True,
+                    'message': 'Action completed successfully'
+                }
+
+        return {
+            'success': True,
+            'message': "",
+            'items': [
+                {
+                    'label': 'Number of tasks',
+                    'type': 'number',
+                    'name': 'number_of_tasks',
+                    'value': 2
+                }
+            ]
+        }
+
+    def _interface(self, *args):
+        interface = self.interface(*args)
+
+        if interface:
+            return {
+                'items': interface
+            }
+
+    def interface(self, session, uid, entities, source, values, event):
```

Comment:
None of the arguments to this method are used. Why use a method that returns nothing?